spec/light-client/supervisor/supervisor_001_draft.md

```diff
@@ -463,69 +463,94 @@ func Sequential-Supervisor (initdata LCInitData) (Error) {
-initialize the lightstore with this first block. The reason is that
-we want to maintain [LCV-INV-TP.1] from the beginning.
-
-> If the lightclient is initialized with a lightblock, one might think
-> it may increase trust, when one cross-checks the initial light
-> block. However, if a peer provides a conflicting
-> lightblock, the question is to distinguish the case of a
-> [bogus](https://informal.systems) block (upon which operation should proceed) from a
-> [light client attack](https://informal.systems) (upon which operation should stop). In
-> case of a bogus block, the lightclient might be forced to do
-> backwards verification until the blocks are out of the trusting
-> period, to make sure no previous validator set could have generated
-> the bogus block, which effectively opens up a DoS attack on the lightclient
-> without adding effective robustness.
+trust the initial data given to it by the user. It cannot perform any
+detection of an attack yet instead requires an initial point of trust.
+There are three forms of initial data:
+
+- A trusted block from a prior initialization
+- A trusted height and hash
+- A genesis file
+
+The golang light client implementation checks this initial data in that 
+order; first attempting to find a trusted block from the trusted store, 
+then acquiring a light block from the primart at the trusted height and matching
+the hash, or finally checking for a genesis file to verify the initial header.
+
+The light client doesn't need to check if the trusted block is within the
+trusted period because it already trusts it, however, if the light block is 
+outside the trust period, there is a higher chance the light client won't be
+able to verify anything. 
+
+Cross-checking this trusted block with providers upon initialization is helpful
+for ensuring liveness and correct configuration of such nodes but does not 
+increase trust since proving a conflicting block is a 
+[light client attack](https://informal.systems) 
+and not just a [bogus](https://informal.systems) block could result in
+performing backwards verification beyond the trusted period, thus a fruitless
+endeavour.
+
+However, with the notion of it's better to fail earlier than later, the golang
+light client implementation will perform a consistency check on all providers
+and will error if one returns a different header, allowing the user
+the opportunity to reinitialize.
 
 #### **[LC-FUNC-INIT.1]:**
 
 ```go
-func InitLightClient (initData LCInitData) (LightStore, Error) {
+func InitLightClient(initData LCInitData) (LightStore, Error) {
+    var initialBlock LightBlock
 
-    if LCInitData.LightBlock != nil {
-        // we trust the provided initial block.
-        newblock := LCInitData.LightBlock
-    }
-    else {
-        genesisBlock := makeblock(initData.genesisDoc);
+    switch {
+    case LCInitData.TrustedBlock != nil:
+        // we trust the block from a prior initialization
```

Comment:
Is it necessarily from "prior" initialization?

Reply:
I guess not. As in it's feasible that someone adds a light block to the store and then passes that in (without it going through a previous initialization process)

Reply:
I just wondered, because the text of the comment was different in the previous version of the spec?

Reply:
Right. The way the light client is initialised, at least in the golang repo, is that we give users the option to either start form a trusted hash and height or from a trusted store which would've come from previously using the light client. I guess I was just translating that notion across

Reply:
It would be great to add this explanation as a comment to the spec.